# GtkFileChooserWidget3: allow to use IEC units in GTK (GTK_USE_IEC_UNITS=1)
# https://github.com/GNOME/gtk/blob/3.24.41/gtk/gtkfilechooserwidget.c
Index: b/gtk/gtkfilechooserwidget.c
===================================================================
--- a/gtk/gtkfilechooserwidget.c
+++ b/gtk/gtkfilechooserwidget.c
@@ -5223,6 +5223,8 @@ file_system_model_set
     case MODEL_COL_SIZE_TEXT:
       if (info == NULL || _gtk_file_info_consider_as_directory (info))
         g_value_set_string (value, NULL);
+      else if (g_strcmp0 (g_getenv ("GTK_USE_IEC_UNITS"), "1") == 0)
+        g_value_take_string (value, g_format_size_full (g_file_info_get_size (info), G_FORMAT_SIZE_IEC_UNITS));
       else
         g_value_take_string (value, g_format_size (g_file_info_get_size (info)));
       break;
